frontend/model/contracts/group.js

```diff
@@ -256,6 +334,46 @@ DefineContract({
         archiveProposal(state, data.proposalHash)
       }
     },
+    'gi.contracts/group/removeMember': {
+      validate: (data) => {
+        return objectOf({
+          member: string,
+          memberID: string,
+          groupID: string,
+          // In case it happens in a big group (by proposal)
+          // we need to validate the associated proposal.
+          proposalHash: optional(string),
+          proposalPayload: optional(objectOf({
+            secret: string // NOTE: simulate the OP_KEY_* stuff for now
+          }))
+        })
+      },
+      process (state, { data, meta }) {
+        const rootState = sbp('state/vuex/state')
+        if (data.member === rootState.loggedIn.username) {
+          // If this member is re-joining the group, ignore the rest
+          // so the member doesn't remove themself again.
+          if (sbp('okTurtles.data/get', 'JOINING_GROUP')) {
+            return false
+          }
+        }
+
+        // QUESTION: Shouldn't these validatons be on validate() method above?
+        // But there we don't have access to state and meta.
+
+        // If we do a validation here and it fails,
+        // will this try to ban/propose to remove only "meta.username" <-- i guess!
+        // or everyone in the group who proccesses this? <-- I saw this happening O.o
+
+        // Well... meanwhile I added validateRemoveMember
+        // on the component side too, before group/removeMember.
+        validateRemoveMember(state, { data, meta })
```

Comment:
You're absolutely right, validations should be done in the `validate` function, not in the `process` function. This is so that developers get automatic validation upon creation of a message, and upon receiving it for processing, without having to write extra lines of code like the call to `validateRemoveMember` in `RemoveMember.vue`.

You are also right, that it's a problem currently that the `validate` function is only passed the `message.data` and is missing the contract state and the metadata. This is a bug that I need to fix.

So for now in this PR, can you leave the `validateRemoveMember` defined, but don't use it anywhere? I've created an issue and self-assigned: #836 — feel free to mention that issue in a comment here in the code.